stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Initializes a vImage buffer of a specified size without memory allocation,
+    /// reporting the preferred alignment.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter alignment: The preferred alignment is written to this parameter.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 alignment: inout Int) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        let error = vImageBuffer_Init(&self,
+                                      height,
+                                      width,
+                                      bitsPerPixel,
+                                      vImage_Flags(kvImageNoAllocate))
+        
+        if error < kvImageNoError {
+            return nil
+        } else {
+            alignment = error
+        }
+    }
+    
+    /// Initializes a vImage buffer of a specified size, reporting any errors.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 error: inout Int,
+                 flags options: vImage.Options = .noFlags) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            error = kvImageInvalidParameter
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        error = vImageBuffer_Init(&self,
+                                  height,
+                                  width,
+                                  bitsPerPixel,
+                                  options.flags)
+        
+        if error < kvImageNoError {
+            return nil
+        }
+    }
+    
+    /// Initializes a vImage buffer of a specified size.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 flags options: vImage.Options = .noFlags) {
```

Comment:
I think that this is the only `init?` you should actually be exposing in the overlay, except that it probably shouldn't have `flags` (as noted above) and should probably take `width` and `height` as `Int` (unless there's a specific reason for `CGSize`).

Reply:
Flags have gone and `size` replaced by `width` and `height`.